src/map/ai/controllers/mob_controller.cpp

```diff
@@ -720,6 +720,26 @@ void CMobController::FaceTarget(uint16 targid)
     }
 }
 
+bool CMobController::IsStuck()
+{
+    return m_Stuck;
+}
+
+void CMobController::UpdateLastKnownPosition()
+{
+    // Mob is considered "Stuck" if:
+    // 1. Last Pos && Current Pos are <= 1.5
+    // 2. Distance to Target > Melee Range
+    // 3. Mob is not bound or asleep
+    m_Stuck =
+        distanceSquared(m_LastPos, PMob->loc.p) <= 1.5f &&
+        distanceSquared(PMob->loc.p, PTarget->loc.p) > PMob->GetMeleeRange() &&
+        PMob->StatusEffectContainer->GetStatusEffect(EFFECT_BIND) == nullptr &&
+        PMob->StatusEffectContainer->GetStatusEffect(EFFECT_SLEEP) == nullptr &&
```

Comment:
There's an IsAsleep() function that includes EFFECT_LULLABY; also do Petrification, Stun, Terror need to come into consideration here as well?
Also, some mobs have a flag that tells them to hang back when casting (I believe `MOBMOD_STANDBACK_COOL`)

Reply:
Cool, those are cases in which I didn't think about.  Since this check is nested deep in `Move()`, we already know the mob should be moving into melee range because `CanMoveForward()` has been called which checks for that specific thing.

Reply:
That if is getting a little wild, so I'll probably just put those checks into another "CanMove" function or something like that.

Reply:
Those changes have now been implemented. Moved to it's on function, also realized you can pass multiple status effects in with a list initializer, so changed to checking that way.